### PR TITLE
plan9port: fix linker flags for macosx

### PR DESCRIPTION
**Pull request by eraserhd**
###### Motivation for this change

PR #77632 made Mac builds more reproducible by adding some linker
flags to `$NIX_LDFLAGS`; however, plan9port uses the C compiler
front end as a linker, requiring these new options to be wrapped
with `-Wl`.

This fixes the Mac build of plan9port.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @LnL7 @AndersonTorres @bbarker @ftrvxmtrx @kovirobi

### CODE REVIEW

**Comment by KoviRobi** [comment]
LGTM, I don't have a Mac but it seems to work on Linux without any changes.

File: pkgs/tools/system/plan9port/builder.sh [hunk 1]
```diff
@@ -8,13 +8,20 @@ plan9portLinkFlags()
...
+            linkFlags+=( "-Wl,$1,$2" )
             shift 2
-        else
+            ;;
+        -no_uuid)
+            linkFlags+=( "-Wl,$1" )
+            shift
+            ;;
+        *)
             linkFlags+=( "$1" )
```
Comment (LnL7) [inline pkgs/tools/system/plan9port/builder.sh]: All the `LDFLAGS` should be sent to the linker instead just handling a few special cases.

Comment (eraserhd) [reply pkgs/tools/system/plan9port/builder.sh]: That works and makes it simpler.

Comment (LnL7) [reply pkgs/tools/system/plan9port/builder.sh]: Great, if that works I can't think of any reason not to do it this way.

**Review by LnL7 — APPROVED** [review]

**Comment by AndersonTorres** [comment]
I have no Darwin machine to test it :/

**Pull request merged**